framework/src/governance/lib/datazone/datazone-msk-environment-authorizer.ts

```diff
@@ -0,0 +1,117 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { Duration, RemovalPolicy } from 'aws-cdk-lib';
+import { IRole, Role, ServicePrincipal, ManagedPolicy, PolicyDocument, PolicyStatement, Effect } from 'aws-cdk-lib/aws-iam';
+import { IFunction, Function, Runtime, Code } from 'aws-cdk-lib/aws-lambda';
+import { IStateMachine } from 'aws-cdk-lib/aws-stepfunctions';
+import { Construct } from 'constructs';
+import { DataZoneMskCentralAuthorizer } from './datazone-msk-central-authorizer';
+import { DataZoneMskEnvironmentAuthorizerProps } from './datazone-msk-environment-authorizer-props';
+import { Context, TrackedConstruct, TrackedConstructProps } from '../../../utils';
+import { authorizerEnvironmentWorkflowSetup } from '../custom-authorizer-environment-helpers';
+
+/**
+ * An environment authorizer workflow for granting read access to Kafka topics.
+ * The workflow is triggered by an event sent by the central authorizer construct.
+ * It creates IAM policies required for the Kafka client to access the relevant topics.
+ * It supports MSK provisioned and serverless, in single and cross accounts, and grant/revoke requests.
+ *
+ * @example
+ * new dsf.governance.DataZoneMskEnvironmentAuthorizer(this, 'MskAuthorizer', {
+ *   domainId: 'aba_dc999t9ime9sss',
+ * });
+ */
+export class DataZoneMskEnvironmentAuthorizer extends TrackedConstruct {
+
+  /**
+   * The IAM role used to grant access to Kafka topics
+   */
+  public readonly grantRole: IRole;
```

Comment:
We need to allow a user to provide this role and we attach a policy to it. If we create it ourselves we need to look at the permission boundary.